rest_framework/serializers.py

```diff
@@ -997,13 +997,19 @@ def update(self, instance, validated_data):
         # relationships as being a special case. During updates we already
         # have an instance pk for the relationships to be associated with.
         m2m_fields = []
+        partial_update_extra_fields = self.get_partial_update_extra_fields()
+        update_fields = [*(partial_update_extra_fields or [])]
         for attr, value in validated_data.items():
             if attr in info.relations and info.relations[attr].to_many:
                 m2m_fields.append((attr, value))
             else:
                 setattr(instance, attr, value)
-
-        instance.save()
+                update_fields.append(attr)
+        
+        if self.partial and partial_update_extra_fields is not None:
```

Comment:
Here it would be better to check only for update fields to be non-empty:

```suggestion
        if self.partial and update_fields:
```

Reply:
I used the condition to avoid changing the current/default behaviour of  `ModelSerializer`. When `partial_update_extra_fields=None` it means `partial_update_extra_fields` was not set therefore `ModelSerializer` should behave as usual. but when `partial_update_extra_fields` is set even like so `partial_update_extra_fields=[]` it should use `update_fields`